python/triton/runtime/jit.py

```diff
@@ -266,6 +266,15 @@ def _type_of(key):
         for v in list(tys.values()):
             tys[v] = v
         return key if isinstance(key, str) else f"*{tys[dtype_str]}"
+    
+    @staticmethod
+    def _is_dtype(key):
```

Comment:
I think we could have something more maintainable here. All the types are already defined in `core.py`.

Reply:
I understand that we would like to reuse them from core.py, but core.py depends on jit.py and that would introduce a cyclic dependency. One suggestion is to move out dtype to a separate file and let both depend on it. What do you think?

Reply:
I was trying out the separation via creating language_types.py as reflected in the current change. If you believe moving @jit'd functions out is better I can change it to reflect what you were saying. Which option do you prefer?

Reply:
Actually I'm more inclined to your suggestion as a better cleanup. I'm updating it now.